docs/blog/_posts/2020-03-18-23rd-dotty-milestone-release.md

```diff
@@ -0,0 +1,289 @@
+
+```scala
+inline def lookup[X] =
+  val x = summon[X]  // error
+  // -- Error: /Users/kmetiuk/Projects/scala3/pg/release/snip_5.scala:6:19 ----------
+  // 6 |  val x = summon[X]
+  //   |                   ^
+  //   |no implicit argument of type X was found for parameter x of method summon in object DottyPredef
+  // 1 error found
+  println(s"x = $x")
+```
+
+The above program will give us a compile time error because it cannot find a context parameter of type `X`. This is because the `summon` function is not inline and hence the compiler needs to know that a context parameter of type `X` exists on call site of `summon` which happens to be in the body of `lookup`. Since `X` is unknown in that body, the compiler can't find the context parameter and shows an error.
+
+We have now added an inline version of `summon`:
+
+```scala
+import scala.compiletime.summonInline
+
+inline def lookup[X] =
+  val x = summonInline[X]
+  println(s"x = $x")
+
+@main def Test =
+  given Int = 10
+  lookup[Int]
+```
+
+`summonInline` is an inline version of `summon`. It is defined as follows:
+
+```scala
+inline def summonInline[T] <: T = summonFrom {
+  case t: T => t
+}
+```
+
+Since it is inline, the context parameter is resolved on expansion site, not the call site. The expansion site happens to be wherever `lookup` function is expanded, and there the type `X` is bound to a concrete type.
+
+## `ValueOfExpr` renamed to `Unlifted`
+This feature allows you to obtain the value captured in an expression using pattern matching:
+
+Macro.scala
+
+```scala
+import scala.quoted._
+
+inline def square(x: Int): Int = ${ squareImpl('x) }
+def squareImpl(x: Expr[Int])(using QuoteContext): Expr[Int] =
+  x match
+    case Unlifted(value: Int) => Expr(value * value)
+```
+
+Test.scala
+
+```scala
+@main def Test =
+  println(square(10))  // 100
+```
+
+## Extractors for quotes moved under `scala.quoted` package
+The metaprogramming capabilities are undergoing simplifications in this release. Particularly, you need to do less imports to use them.
+
+Previously, to access the extractors for expressions you had to do the `scala.quoted.matching._` import. Now, the extractors from there have been moved to `scala.quoted`. For example, you can write the following program:
+
+Macro.scala:
+
+```scala
+import scala.quoted._
+
+inline def square(x: Int): Int = ${ squareImpl('x) }
+def squareImpl(xExpr: Expr[Int])(using QuoteContext): Expr[Int] =
+  xExpr match
+    case Const(x) => Expr(x * x)
+```
+
+Test.scala:
+
+```scala
+@main def Test =
+  println(square(2))  // 4
+```
+
+Above, `Const` is an extractor that matches constants. Notice how we do not need to import anything else but `scala.quoted._` to use it.
+
+## TASTy Reflect imports simplified
+Previously, to access TASTy Reflect features of Dotty, you had to include an import as follows:
```

Comment:
Maybe there should be a reference to the changes to implicit scope rules that allow this to happen